src/main/java/deez/Ui.java

```diff
@@ -0,0 +1,59 @@
+package deez;
+
+import java.util.ArrayList;
+import java.util.function.Consumer;
+
+/**
+ * The Ui class contains utility methods for user interface.
+ */
+public class Ui {
+    // TODO: We should probably abstract the payload of this consumer to support
+    //       a more diverse range of actions
+    private Consumer<String> messageConsumer;
+
+    /**
+     * Initialise Ui in a headless manner
+     */
+    public Ui() {
+        messageConsumer = (String message) -> {
+            System.out.println("WARN: Ui is in headless mode");
+            System.out.println(message);
+        };
+    }
+
+    /**
+     * Initialise Ui with a message consumer
+     */
+    public Ui(Consumer<String> messageConsumer) {
```

Comment:
The JavaDoc for this constructor should explain what the messageConsumer does with a `@params` tag 😄